File: C#/Medium/Coin Change.py
class Solution:
    def coinChange(self, coins: List[int], amount: int) -> int:
        d = [amount+1] * (amount+1)
        d[0] = 0
        for i in range(amount + 1):
            for c in coins:
                if i >= c:
                    d[i] = min(d[i], 1 + d[i-c])
        
        return d[amount] if d[amount] < amount+1 else -1

File: LeetCode/C#/Medium/Coin Change.cs
public class Solution {
    public int CoinChange(int[] coins, int amount) {
        int[] dp = new int[amount+1];
        Array.Fill(dp, amount+1);
        dp[0] = 0;
        
        for(int i=1 ;i< dp.Length; i++) {
            foreach(int j in coins) {
                if(i >= j) {
                    dp[i] = Math.Min(dp[i], dp[i-j] + 1);
                }
            }
        }
        return dp[amount] == amount+1 ? -1 : dp[amount];
    }
}